numba/cuda/tests/cudapy/test_casting.py

```diff
@@ -64,12 +104,116 @@ def test_float_to_unsigned(self):
         self.assertEqual(cfunc(3.21),
                          struct.unpack('I', struct.pack('i', 3))[0])
 
-    def test_float_to_complex(self):
+    def test_float16_to_unsigned(self):
+        pyfunc = float_to_unsigned
+        cfunc = self._create_wrapped(pyfunc, np.float16, np.uint32)
+
+        self.assertEqual(cfunc(3.21), pyfunc(3.21))
+        self.assertEqual(cfunc(3.21),
```

Comment:
I think this might be to get the wraparound for negative numbers in unsigned domain:

```python
>>> struct.unpack("I", struct.pack("i", 3))[0]
3
>>> struct.unpack("I", struct.pack("i", -3))[0]
4294967293
```

Reply:
I think we saw platform-specific differences doing that (https://github.com/numba/numba/pull/1048) because it wasn't defined behaviour.